### DISCUSSION TITLE
Visual Studio support

### DISCUSSION THREAD

**Discussion by asasas9500** [LostArtefacts/TRX#375] [Ideas]
I think that an alternative build workflow using Visual Studio (and maybe CMake) would be very useful for Windows contributors.

**Comment by walkawayy** [comment]
I think this could be useful since a lot of potential contributors are building on Windows environments since it's a (for now) Windows only game. Deferring to @rr- though since he is in charge of the project / building pipelines.

**Reply by rr-** [reply]
I don't have Visual Studio and I will not use it in any foreseeable future. I am personally completely unwilling to maintain it.

If there can be an easy setup that integrates libavcodec and SDL and does not require hardcoding paths to anything, I will merge it. Ideally something that can be integrated with GitHub actions to immediately see when we break the build.

**Reply by asasas9500** [reply]
SDL configure step rewrites Makefile.in into Makefile (without .in extension). This is the relevant line: https://github.com/rr-/Tomb1Main/blob/master/docker/Dockerfile#L90. Can you post its contents? It will be useful to understand the install target. walkawayy tried to help me but we were not successful.

**Reply by walkawayy** [reply]
I'm a Docker noob lol. In the dockerfile there are a few commands to get dependencies like the `RUN git clone https://github.com/libsdl-org/SDL` line. But, those folders aren't cloned into the container it seems, and when I inspected the layers I don't see the steps for that command. Does it just clone, install, then delete the cloned SDL repo?

**Reply by rr-** [reply]
We're using multistage builds so that we don't have tons of leftovers from building the binary artifacts. The final result gets persisted as a pre-built image on the DockerHub, which GitHub Actions and contributors later pull and build T1M with, without having to waste time on recompiling all dependencies from scratch.

Making individual people suffer with building SDL and ffmpeg themselves is not the way to go. Doesn't Visual Studio have something like NuGet for C++?… (If not, what's so great about it?)

**Reply by asasas9500** [reply]
Can you change the script to keep the leftovers?

**Reply by rr-** [reply]
Not really. If you absolutely do require that, you can change Dockerfile by yourself; just remove the `FROM` lines to get a single-stage build and tweak it until it builds. But I'm not sure why you need this. Like I said, forcing people to compile these big dependencies (which takes about half an hour on a high end CPU) is not the way to go. Shipping binary DLLs or .LIB for SDL in this repository is also undesirable.

I think you can see why I don't want to have anything to do with Visual Studio.

**Comment by rr-** [comment]
Offtopic for the people who don't want to "install Linux" on their Windows machines: I don't think you even need WSL, you can try your luck with this https://docs.docker.com/desktop/windows/install/
I don't have Windows so I can't guide you step by step. For exact command line refer to the Makefile.

**Reply by walkawayy** [reply]
Docker for Windows system requirements are WSL 2 backend or Hyper-V backend and Windows containers (which are Windows 10 only).

Hopefully @asasas9500 will be able to figure out the Windows (non WSL) + VS build route. It's not your preferred method (or mine), but it seems like other people would like the option.
